azurelinuxagent/common/event.py

```diff
@@ -128,9 +139,45 @@ def _save(self):
     ]
 
 
+def _encode_message(op, message):
+    """
+    Gzip and base64 encode a message based on the operation.
+
+    The intent of this message is to make the logs human readable and include the
+    stdout/stderr from extension operations.  Extension operations tend to generate
+    a lot of noise, which makes it difficult to parse the line-oriented waagent.log.
+    The compromise is to encode the stdout/stderr so we preserve the data and do
+    not destroy the line oriented nature.
+
+    The data can be recovered using the following command:
+
+      $ echo '<encoded data>' | base64 -d | pigz -zd
+
+    You may need to install the pigz command.
```

Comment:
Consider having the guest agent, at startup, log this brief summary of "how to read the stdout/stderr blob" to the agent log. That may help forestall a few support calls.

Reply:
I don't think a startup message is the appropriate spot for this, but I am happy to add it to the wiki.

Reply:
That works. Discoverability is key.